colossalai/inference/tensor_parallel/engine.py

```diff
@@ -124,6 +181,13 @@ def _shard_model_by(self, shardformer: ShardFormer, model: nn.Module) -> None:
         model_name = model.__class__.__name__
         assert model_name in self.supported_models, f"Unsupported model cls {model_name} for TP inference."
         policy = get_autopolicy(model, inference_only=True)
+        if not hasattr(policy, "gptq"):
```

Comment:
Same problem as above. It seems to be a hack to do optimization explicitly at engine level. It's better to execute this step in policy (e.g. shard former replacing layers) based on an option of whether to do gptq.